cli/src/config.ts

```diff
@@ -57,6 +54,86 @@ export async function loadConfig(): Promise<Config> {
   return config;
 }
 
+type ExtConfigPairs = Pick<
+  AppConfig,
+  'extConfigType' | 'extConfigName' | 'extConfigFilePath' | 'extConfig'
+>;
+
+async function loadExtConfigTS(
+  rootDir: string,
+  extConfigName: string,
+  extConfigFilePath: string,
+): Promise<ExtConfigPairs> {
+  try {
+    const tsPath = resolveNode(rootDir, 'typescript');
+
+    if (!tsPath) {
+      logFatal(
+        'Could not find installation of TypeScript.\n' +
+          `To use ${c.strong(
+            extConfigName,
+          )} files, you must install TypeScript in your project, e.g. w/ ${c.input(
+            'npm install -D typescript',
+          )}`,
+      );
+    }
+
+    const ts = require(tsPath); // eslint-disable-line @typescript-eslint/no-var-requires
+
+    return {
+      extConfigType: 'ts',
+      extConfigName,
+      extConfigFilePath: extConfigFilePath,
+      extConfig: requireTS(ts, extConfigFilePath) as any,
```

Comment:
I think it'd be quite useful to optionally support exporting a `() => Promise<ExternalConfig>` instead of just `ExternalConfig`.

Would you accept a pull request?

Reply:
Yes, I was thinking about this as well. I would be happy to look at a PR that adds that functionality.